### DISCUSSION TITLE
How the Jacobian is defined in Moose

### DISCUSSION THREAD

**Discussion by Oops-Qiao** [idaholab/moose#21876] [Q&A General] [answered]
Hello,

Today I just read something in the following link
https://mooseframework.inl.gov/syntax/Kernels/index.html#824e4ebf-e857-4e31-9fbf-290950ee3314
where it mentioned:  The Jacobian, which is the derivative of [Eq. (2)] ...

I am curious if this is the same way as we studied finite element method course. Or this is just a special way for Jacobian in Moose.

Thanks for your answers.

Best regards,
Qiao

**Comment by Oops-Qiao** [comment]
If I remember correctly, Jacobian is defined as the derivative of physical domain coordinates with respect to the reference domain.
Why the derivative of Eq 2 is equivalent to this Jacobian mapping?

**Comment by hugary1995** [comment, marked as answer]
Yeah, I think this confuses many people.

Wikipedia says this: https://en.wikipedia.org/wiki/Jacobian_matrix_and_determinant

So Jacobian is a general definition. The three most common "specializations" you may find in MOOSE or any FEM packages are
1. The derivative of the residual vector w.r.t. all dofs
2. The derivative of the element coordinates w.r.t. the reference element, in the context of parametric mapping.
3. The derivative of the coordinates in the current/deformed configuration (after applying displacements) w.r.t. the coordinates in the reference configuration, in the context of mechanics.

**Reply by hugary1995** [reply]
Yeah, sorry I don't check my school emails very often. I just updated my github profile.
